src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java

```diff
@@ -1,7 +1,240 @@
 package pl.byczazagroda.trackexpensesappbackend.controller;
 
-import static org.junit.jupiter.api.Assertions.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.service.WalletServiceImpl;
 
+import java.time.Instant;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = WalletController.class,
+        excludeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = WalletServiceImpl.class),
+        includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {
+                WalletModelMapper.class}))
 class WalletControllerTest {
 
+    @MockBean
+    private WalletServiceImpl walletService;
+
+    @Autowired
+    private WalletController underTest;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void itShouldReturnStatusOKAndCorrectResponseBody() throws Exception {
+        //GIVEN
+        Instant timeCreated = Instant.now();
+
+        TestEditWalletDto editWalletDto = returnTestCorrectEditWalletDto();
+        given(walletService.updateWallet(Mockito.any())).willReturn(
+                new WalletDTO(
+                        editWalletDto.getId(),
+                        editWalletDto.getName(),
+                        timeCreated
+                )
+        );
+
+        //WHEN
+        ResultActions editResultActions = mockMvc.perform(
+                patch("/api/wallet/edit")
```

Comment:
In Controller layer we defined put rest api method with `/api/wallet/update` endpoint but in `WalletControllerTest.java` we test patch method with `/api/wallet/edit` endpoint. It should be corrected

Reply:
Updated.

Reply:
Have you ever heard about "Richardson Maturity Model"? We implement Rest Architecture and this mean that we use "Richardson Maturity Model". For our project we use level 2: https://martinfowler.com/articles/richardsonMaturityModel.html#level2 . According to this cannot use verbs as an endpoint (edit is a verb). Please remove "/edit".  Additionally please read more about "Richardson Maturity Model"